### DISCUSSION TITLE
reading from novelai image not working?

### DISCUSSION THREAD

**Discussion by Weroxig** [receyuki/stable-diffusion-prompt-reader#71] [Q&A] [answered]
test image: https://desu-usergeneratedcontent.xyz/g/image/1712/50/1712500928517.png
works on https://novelai.net/inspect but doesn't work on this tool for me.
![image](https://github.com/receyuki/stable-diffusion-prompt-reader/assets/105804584/992b0fce-b01a-40e6-8fad-4f2de6ad97fc)

**Comment by receyuki** [comment]
This is incredible! There's no metadata in this image, but NovelAI's tool can still display the prompt. 
My guess is when you use NovelAI to generate a image, NovelAI records the checksum of the image, along with its metadata, and provides a image without any metadata for you to download. When you use NovelAI's tools to recognise metadata, it will query the server's records. Therefore, I won't be able to retrieve this metadata locally.

**Reply by Weroxig** [reply]
They store information in the alpha channel
https://github.com/NovelAI/novelai-image-metadata

**Reply by receyuki** [reply]
This is very interesting; I will try to add it to the parser in the next release.

**Comment by receyuki** [comment, marked as answer]
Please try the new beta version, I have added a new parser for novelai.
https://github.com/receyuki/stable-diffusion-prompt-reader/releases/tag/v1.3.5-beta.1

**Comment by albertlast** [comment]
Found on Redit also this style but the images are in webp: https://preview.redd.it/whats-that-something-furry-is-approaching-v0-drna37hcb9vc1.png?width=1080&crop=smart&auto=webp&s=62c432623502f5050a30958aa30719c1386dd3d5
When a random image converter is taken and convert to png (RGBA),
the image information can be read by the same logic.

**Reply by receyuki** [reply]
The image from this link might have been compressed, so novelai.net/inspect doesn't work. Do you still have the link to the original Reddit post?
<img width="806" alt="截屏2024-05-07 下午4 52 41" src="https://github.com/receyuki/stable-diffusion-prompt-reader/assets/28808141/655b065e-93eb-42fe-a6a0-62a5237543e6">

**Reply by albertlast** [reply]
maybe i got confused here with a image, this works for me:
https://www.reddit.com/media?url=https%3A%2F%2Fpreview.redd.it%2Fjmgk5eooiiyc1.png%3Fwidth%3D1024%26format%3Dpng%26auto%3Dwebp%26s%3D8d0d7b56a228bd157401286e16c3ebac797e3cd8
![grafik](https://github.com/receyuki/stable-diffusion-prompt-reader/assets/1782906/d3cbf1a1-594f-4b09-832b-133f198dada5)

**Reply by receyuki** [reply]
Ok, I will add support for webp in beta2. I just didn't have any novelai samples in webp format before, so thank you so much for your sample.

**Reply by receyuki** [reply]
Beta2 is here! I've added support for webp, please help me test any new images you find.
https://github.com/receyuki/stable-diffusion-prompt-reader/releases/tag/v1.3.5-beta.2

**Reply by receyuki** [reply]
@albertlast
